### ISSUE TITLE
Copilot Vault QA reports error on Orama database connection error resulting in Vault QA not working as expected. 

### ISSUE THREAD

**Issue by MarkVerbeek91** [logancyang/obsidian-copilot#809]
- [x] Disable all other plugins besides Copilot **(required)**
- [x] Screenshot of note + Copilot chat pane + dev console added **(required)**

Copilot version: 2.7.1

(Bug report without the above will be closed)

**Describe how to reproduce**
Have copilot enabled as the only community plug-in, load obsidian, wait for a half a minute or so. The following error will appear: `Error initializing Orama database: Error: Connection error.`

I have LM studio installed with the `llama-3.2-1b-instruct` model running. The `chat` option does work as expected. 

**Expected behavior**
First of all I would not expect an error message to appear. But mostly the `vault QA` does not give any answer, thus also no answers based on my notes. 

**Screenshots**
Error at obsidian and copilot start:
![image](https://github.com/user-attachments/assets/67b399bb-ca91-4bc4-a18b-66f4b953f8ff)

Error at switch from `chat` to `Vault QA` and pressing the `refresh index` button. 
![image](https://github.com/user-attachments/assets/86a34cc1-694d-4fd7-bc4f-6930b46608e5)

Asking a question in `Vault QA`:
![image](https://github.com/user-attachments/assets/48da001e-b37c-43af-b122-a07b6b44503a)

**Additional context**
I'm running obsidian 1.7.6 on windows 11. 

some LM studio logs:

`
2024-11-13 21:28:17 [DEBUG]
llama_perf_context_print:        load time =    5740.67 ms
llama_perf_context_print: prompt eval time =      95.52 ms /   347 tokens (    0.28 ms per token,  3632.59 tokens per second)
llama_perf_context_print:        eval time =    1401.03 ms /    94 runs   (   14.90 ms per token,    67.09 tokens per second)
llama_perf_context_print:       total time =    1573.84 ms /   441 tokens
2024-11-13 21:28:17  [INFO] Finished streaming response
2024-11-13 21:28:17  [INFO] [LM STUDIO SERVER] Client disconnected. Stopping generation..
2024-11-13 21:28:17  [INFO]
Received POST request to /v1/chat/completions with body: {
  "model": "llama-3.2-1b-instruct",
  "temperature": 0.1,
  "top_p": 1,
  "presence_penalty": 0,
  "n": 1,
  "stream": true,
  "stream_options": {
    "include_usage": true
  },
  "messages": [
    {
      "role": "user",
      "content": "Human: Please write a passage to answer the question. If you don't know the answer, just make up a passage. \nQuestion: Here is the summary of the conversation as context:\n\nThe conversation started with a discussion about the time it takes for a lead ball and a feather to reach the ground when dropped simultaneously from the same height, based on their initial conditions and the laws of gravity. The follow up input was then provided, asking what color a golden ring might be.\n\nIf the follow up question is unrelated to its preceding messages, return this follow up question directly:\n\nWhat color is a golden ring?\nPassage:"
    }
  ]
}
2024-11-13 21:28:17  [INFO] [LM STUDIO SERVER] Running chat completion on conversation with 1 messages.
2024-11-13 21:28:17  [INFO] [LM STUDIO SERVER] Streaming response...
2024-11-13 21:28:17 [DEBUG]
sampling: 
logits -> logit-bias -> penalties -> top-k -> tail-free -> typical -> top-p -> min-p -> temp-ext -> softmax -> dist 
generate: n_ctx = 4096, n_batch = 512, n_predict = -1, n_keep = 161
2024-11-13 21:28:17  [INFO] [LM STUDIO SERVER] First token generated. Continuing to stream response..
2024-11-13 21:28:20 [DEBUG]
llama_perf_context_print:        load time =    5740.67 ms
llama_perf_context_print: prompt eval time =      35.82 ms /   130 tokens (    0.28 ms per token,  3629.26 tokens per second)
llama_perf_context_print:        eval time =    2862.87 ms /   207 runs   (   13.83 ms per token,    72.31 tokens per second)
llama_perf_context_print:       total time =    3069.51 ms /   337 tokens
2024-11-13 21:28:20  [INFO] Finished streaming response
2024-11-13 21:28:20  [INFO] [LM STUDIO SERVER] Client disconnected. Stopping generation..`

**Comment by logancyang** [comment]
What's your embedding model? Looks like your embedding model isn't working.

Documentation at https://obsidiancopilot.com

**Comment by MarkVerbeek91** [comment]
I used the same model as for the chat option, reading the documentation again I now understand that I need a special embeddings model. Is it possible to use LM studio for this because only Ollama is mentioned as an option for local use.

**Comment by logancyang** [comment]
> I used the same model as for the chat option, reading the documentation again I now understand that I need a special embeddings model. Is it possible to use LM studio for this because only Ollama is mentioned as an option for local use.

Yes you can, just search for LM studio in issues and you will see how people are using it for embedding. Last time I tried, LM studio's embedding model has problem with silently failing so I did not add it to the official list of supported embedding models. That may have changed on the LM studio end since then.

**Comment by MarkVerbeek91** [comment]
Thanks for the support. I have now a working setup. 

From documentation of LM studio it was not very clear that it only could do load LLMS or a single embedding model. I found this page that mentions this fact: [AnythinkLLM](https://docs.useanything.com/setup/embedder-configuration/local/lmstudio)

Copilot is now working with LM Studio for my LLM and Ollama for the embeddings models (I used `nomic-embed-text`)

Perhaps the documentation of Copilot could be augmented with this information.
